kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/converters/impl/AttachmentConverter.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.kotlindiscord.kord.extensions.commands.converters.impl
+
+import com.kotlindiscord.kord.extensions.commands.Argument
+import com.kotlindiscord.kord.extensions.commands.CommandContext
+import com.kotlindiscord.kord.extensions.commands.converters.SingleConverter
+import com.kotlindiscord.kord.extensions.commands.converters.Validator
+import com.kotlindiscord.kord.extensions.modules.annotations.converters.Converter
+import com.kotlindiscord.kord.extensions.modules.annotations.converters.ConverterType
+import com.kotlindiscord.kord.extensions.parser.StringParser
+import dev.kord.core.entity.Attachment
+import dev.kord.core.entity.interaction.AttachmentOptionValue
+import dev.kord.core.entity.interaction.OptionValue
+import dev.kord.rest.builder.interaction.AttachmentBuilder
+import dev.kord.rest.builder.interaction.OptionsBuilder
+
+/**
+ * Argument converter for Discord attachments.
+ */
+@Converter(
+    "attachment",
+
+    types = [ConverterType.OPTIONAL, ConverterType.SINGLE],
+)
+public class AttachmentConverter(
+    override var validator: Validator<Attachment> = null
+) : SingleConverter<Attachment>() {
+    override val signatureTypeString: String = "converters.attachment.signatureType"
+
+    override suspend fun parse(parser: StringParser?, context: CommandContext, named: String?): Boolean = false
```

Comment:
This needs a more descriptive, translated error - thrown in a `DiscordRelayedException`. I think there's already a translation string you can use in there, but I'm not sure.

Reply:
I added a DiscordRelayedException, didn't find a translation string for it, added a new one.

I found a very similar case, for `ChoiceConverters`, when trying to use one in a chat command, it throws an IllegalStateException:
```kt
is ChoiceConverter<*> -> error("Choice converters may only be used with slash commands")
```
https://github.com/Kord-Extensions/kord-extensions/blob/develop/kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/chat/ChatCommandParser.kt#L119